### PR TITLE
Add change management section to regulatory docs

### PR DESCRIPTION
**Pull request by atrigila**
Expand change management section explaining how nf-core handles PRs. 

@netlify /docs/guidelines/regulatory/overview

### CODE REVIEW

File: sites/docs/src/content/docs/guidelines/regulatory/overview.md [hunk 1]
```diff
@@ -113,7 +112,16 @@ Make sure you rerun the appropriate levels of testing depending on the potential
...
+In software development, change management refers to the process of tracking and controlling modifications to code and documentation throughout the _software development lifecycle_, ensuring transparency, accountability, and risk mitigation. In regulated bioinformatics environments, change management practices are essential for maintaining compliance, and reproducibility. Within nf-core pipelines, change management is structured as follows:
+
+- Requests for changes, bug reports, and enhancement suggestions can be submitted by any user or community member, ensuring transparent and open community-driven improvement.
+- Each pipeline and module follows a Pull Request (PR) template checklist, which helps contributors meet minimum submission requirements.
+- Proposed changes must include system and unit tests, which are automatically validated through the continuous integration/continuous deployment (CI/CD) framework, reducing manual testing overhead.
+- Changes to development branches require peer review, with each PR needing at least one review before merging into dev branches and at least two reviews before merging into main/master.
```
Comment (tomiles) [inline sites/docs/src/content/docs/guidelines/regulatory/overview.md:120]:
```suggestion
- Changes to development branches require peer review, with each PR needing at least one review before merging into dev branches and at least two reviews before merging into the _main_ branch.
```

File: sites/docs/src/content/docs/guidelines/regulatory/overview.md [hunk 2]
```diff
@@ -113,7 +112,16 @@ Make sure you rerun the appropriate levels of testing depending on the potential
...
+- Automated tests are triggered on each PR to confirm that existing functionality remains unaffected.
+- Automated linting checks are performed on each PR, enforcing coding standards and preventing stylistic issues.
+- Direct changes to the master branch are not permitted, protecting the integrity of the production-ready code.
```
Comment (tomiles) [inline sites/docs/src/content/docs/guidelines/regulatory/overview.md:123]:
```suggestion
- Direct changes to the main branch are not permitted, protecting the integrity of the production-ready code.
```

**Review by tomiles — COMMENTED** [review]
Looks good, although I'm not sure how specific we want to go as specific for the **Change Management** section, some items seem more in the category of "nice to have" but are not really a regulatory necessity. 

(added some small fixes so we consistently use `main` branch as the primary branch label.

**Review by apeltzer — APPROVED** [review]

**Comment by apeltzer** [comment]
@nf-core-bot fix linting

**Comment by apeltzer** [comment]
@nf-core-bot fix linting

**Comment by atrigila** [comment]
Thanks for your feedback, you are right, I added a small note on how this relates to re-validation of pipelines.

**Review by tomiles — APPROVED** [review]

**Pull request merged**